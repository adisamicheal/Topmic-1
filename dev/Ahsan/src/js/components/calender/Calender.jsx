import React from 'react';
import Content from './Content';
import events from './Data'

export default class Calender extends React.Component {
	constructor(){
		super();
		
		this.state={
			n:0
		}
		this.onClickRightArrow=this.onClickRightArrow.bind(this);
		this.onClickLeftArrow=this.onClickLeftArrow.bind(this);
	}

	onClickLeftArrow(){
		let {n}=this.state
		if(n<=0){
			return this.setState({
				n:0
			})
		}
		else{
			return this.setState({
				n:n-1
			})
		}	
	}
	onClickRightArrow(){
		let {n}=this.state
		if(this.state.n >= events.length-4){
			return this.setState({
				n:events.length-4
			})
		}
		else{
			return this.setState({
				n:n+1
			})
		}	
	}
	render(){
		const {n}=this.state;
		return (
			<div>
			<Content
			 n={n}
			 onClickLeftArrow={this.onClickLeftArrow}
			 onClickRightArrow={this.onClickRightArrow}
			 />	
			</div>
		)
	}
} 