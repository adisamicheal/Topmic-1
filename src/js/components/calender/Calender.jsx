import React from 'react';
import MediaQuery from 'react-responsive';
import Content from './Content';

const Desktop = props => (<MediaQuery {...props} maxWidth={1400} minWidth={992} />);
const Tablet = props => <MediaQuery {...props} minWidth={768} maxWidth={991} />;
const Mobile = props => <MediaQuery {...props} maxWidth={767} minWidth={600} />;
const SmallMobile = props => (<MediaQuery {...props} maxWidth={601} minWidth={468} />);
const ExtraSmallMobile = props => <MediaQuery {...props} maxWidth={468} />;
const Calender = () => (
	<div>
		<Desktop>
			<Content count={4} />
		</Desktop>
		<Tablet>
			<Content count={3} />
		</Tablet>
		<Mobile>
			<Content count={2} />
		</Mobile>
		<SmallMobile>
			<Content count={1} />
		</SmallMobile>
		<ExtraSmallMobile>
			<Content count={1} />
		</ExtraSmallMobile>
	</div>
);

export default Calender;
