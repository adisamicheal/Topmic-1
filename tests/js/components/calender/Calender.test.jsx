import React from 'react';
import Enzyme , {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Calender from '../../../../src/js/components/calender/Calender';

Enzyme.configure({adapter:new Adapter()});

describe('Test the Calender Component',()=>{
	const wrapper=shallow(<Calender />)
	it('checks the rendering of calender app component',()=>{
		expect(wrapper).toMatchSnapshot();
	})
})