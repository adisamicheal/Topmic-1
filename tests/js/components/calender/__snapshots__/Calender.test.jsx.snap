// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Calender Component checks the rendering of calender app component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calender />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Desktop>
          <Content
            count={4}
          />
        </Desktop>,
        <Tablet>
          <Content
            count={3}
          />
        </Tablet>,
        <Mobile>
          <Content
            count={2}
          />
        </Mobile>,
        <SmallMobile>
          <Content
            count={1}
          />
        </SmallMobile>,
        <ExtraSmallMobile>
          <Content
            count={1}
          />
        </ExtraSmallMobile>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Content
            count={4}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Content
            count={3}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Content
            count={2}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Content
            count={1}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Content
            count={1}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Desktop>
            <Content
              count={4}
            />
          </Desktop>,
          <Tablet>
            <Content
              count={3}
            />
          </Tablet>,
          <Mobile>
            <Content
              count={2}
            />
          </Mobile>,
          <SmallMobile>
            <Content
              count={1}
            />
          </SmallMobile>,
          <ExtraSmallMobile>
            <Content
              count={1}
            />
          </ExtraSmallMobile>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Content
              count={4}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "count": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Content
              count={3}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "count": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Content
              count={2}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "count": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Content
              count={1}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "count": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Content
              count={1}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "count": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
